service/src/main/kotlin/fi/espoo/evaka/attendance/StaffAttendanceQueries.kt

```diff
@@ -277,13 +278,19 @@ fun Database.Read.getExternalStaffAttendancesByDateRange(unitId: DaycareId, rang
     .mapTo<ExternalAttendance>()
     .list()
 
-fun Database.Read.getGroupsForEmployee(employeeId: EmployeeId): List<GroupId> = createQuery(
+private data class EmployeeGroups(
+    val employeeId: EmployeeId,
+    @Json
+    val groupIds: List<GroupId>
+)
+fun Database.Read.getGroupsForEmployees(employeeIds: Set<EmployeeId>): Map<EmployeeId, List<GroupId>> = createQuery(
     """
-    SELECT daycare_group_id
+    SELECT employee_id, json_agg(daycare_group_id) AS group_ids
```

Comment:
Contrary to what I previously said `array_agg(...)` would probably be more readable here. Then you can also remove the `@Json` annotation above.